File: torneo.py

equipos = ["Colombia", "Nicaragua", "Panama", "España",
           "Mexico", "Alemania", "Venezuela", "Chile",
           "Argentina", "Brazil", "Ecuador", "Japon",
           "Honduras", "Canada", "Francia", "Peru"]

# Función para simular un enfrentamiento entre dos equipos
def enfrentamiento(equipo1, equipo2):
    # indicamos los marcadores de cada equipo
    print ("usuari@ favor recuerde que estos partidos NO empatan")
    print (f"los equipos que se enfrentaran son: {equipo1} y {equipo2}")
    print (f"ingrese el marcador para equipo incluya penales {equipo1}")
    a = int(input())
    print (f"ingrese el marcador para equipo incluya penales {equipo2}")
    b = int(input())
    if a > b:
        ganador = equipo1
    else:
        ganador = equipo2

    print(f"{equipo1} vs {equipo2} -> Gana: {ganador}")
    return ganador

# Función para simular una ronda del torneo
def ronda(equipos):
    ganadores = []
    for i in range(0, len(equipos), 2):
        equipo1 = equipos[i]
        equipo2 = equipos[i + 1]
        ganador = enfrentamiento(equipo1, equipo2)
        ganadores.append(ganador)
    return ganadores

# Simulación del torneo
ronda_actual = equipos
ronda_numero = 1

while len(ronda_actual) > 1:
    print(f"\n--- Ronda {ronda_numero} ---")
    ronda_actual = ronda(ronda_actual)
    ronda_numero += 1

# El último equipo restante es el ganador del torneo
print(f"\n¡El ganador del torneo es {ronda_actual[0]}!")
print("el subcampeon es chile")

File: trabajos en rubi/torneo.rb
def simular_partido(equipo1, equipo2)
  puts "Partido: #{equipo1} vs #{equipo2}"
  print "Ingrese el marcador de #{equipo1}: "
  goles_equipo1 = gets.chomp.to_i
  print "Ingrese el marcador de #{equipo2}: "
  goles_equipo2 = gets.chomp.to_i

  if goles_equipo1 > goles_equipo2
    return equipo1
  elsif goles_equipo2 > goles_equipo1
    return equipo2
  else
    puts "Empate. Definición por penales."
    print "Ingrese el ganador por penales (#{equipo1} o #{equipo2}): "
    return gets.chomp
  end
end

def simular_fase(equipos)
  ganadores = []
  while equipos.length > 0
    equipo1 = equipos.shift
    equipo2 = equipos.shift
    ganador = simular_partido(equipo1, equipo2)
    ganadores << ganador
  end
  return ganadores
end

# Equipos participantes
equipos = ["Colombia", "Panama", "Peru", "Ecuador", "Brasil", "Argentina", "Chile", "Uruguay", "Paraguay", "Venezuela", "Bolivia", "Mexico", "Costa Rica", "Honduras", "Estados Unidos", "Canada", "Jamaica", "Haiti", "Guatemala", "El Salvador", "Nicaragua", "Cuba", "Puerto Rico", "Republica Dominicana", "Trinidad y Tobago", "Barbados", "Granada", "San Vicente y las Granadinas", "Santa Lucia", "Antigua y Barbuda", "San Cristobal y Nieves", "Bahamas", "Belice", "Guyana", "Surinam", "Guyana Francesa", "Brasil", "Argentina", "Chile", "Uruguay", "Paraguay", "Venezuela", "Bolivia", "Mexico", "Costa Rica", "Honduras", "Estados Unidos", "Canada", "Jamaica", "Haiti", "Guatemala", "El Salvador", "Nicaragua", "Cuba", "Puerto Rico", "Republica Dominicana", "Trinidad y Tobago", "Barbados", "Granada", "San Vicente y las Granadinas", "Santa Lucia", "Antigua y Barbuda", "San Cristobal y Nieves", "Bahamas", "Belice", "Guyana", "Surinam", "Guyana Francesa", "Brasil", "Argentina", "Chile", "Uruguay", "Paraguay", "Venezuela", "Bolivia", "Mexico", "Costa Rica", "Honduras", "Estados Unidos", "Canada", "Jamaica", "Haiti", "Guatemala", "El Salvador", "Nicaragua", "Cuba", "Puerto Rico", "Republica Dominicana", "Trinidad y Tobago", "Barbados", "Granada", "San Vicente y las Granadinas", "Santa Lucia", "Antigua y Barbuda", "San Cristobal y Nieves", "Bahamas", "Belice", "Guyana", "Surinam", "Guyana Francesa"]

# Simulación del torneo
fase = 1
while equipos.length > 1
  puts "\n--- Fase #{fase} ---"
  puts "Partidos de la Fase #{fase}:"
  equipos_fase = equipos.dup # Crear una copia para mostrar los partidos
  while equipos_fase.length > 0
    puts "#{equipos_fase.shift} vs #{equipos_fase.shift}"
  end
  equipos = simular_fase(equipos)
  puts "Ganadores de la Fase #{fase}: #{equipos.join(", ")}"
  fase += 1
end

puts "\n¡El ganador del torneo es: #{equipos[0]}!"
puts
print ("fin de la ejecucion_ escriba exit para salir:")
Z = gets.chomp.to_i